File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/course_detail/uploads/data/CourseUpload.java
package org.wikiedufoundation.wikiedudashboard.course_detail.uploads.data;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@AllArgsConstructor
@ToString
public class CourseUpload implements Serializable {
    private int id;
    private String uploaded_at;
    private String usage_count;
    private String url;
    @SerializedName("thumburl")
    private String thumbUrl;
    private boolean deleted;
    @SerializedName("thumbwidth")
    private String thumbWidth;
    @SerializedName("thumbheight")
    private String thumbHeight;
    private String file_name;
    private String uploader;
}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/uploads/data/CourseUpload.kt
package org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.data

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * [CourseUpload] model class
 * @constructor primary constructor to initialize properties and variables
 * ***/
data class CourseUpload  (
    @SerializedName("id")
    val id: Int? = null,
    @SerializedName("uploaded_at")
    val uploadedAt: String? = null,
    @SerializedName("usage_count")
    val usageCount: String? = null,
    @SerializedName("url")
    val url: String? = null,
    @SerializedName("thumburl")
    val thumbUrl: String? = null,
    @SerializedName("deleted")
    val deleted: Boolean? = null,
    @SerializedName("thumbwidth")
    val thumbWidth: String? = null,
    @SerializedName("thumbheight")
    val thumbHeight: String? = null,
    @SerializedName("file_name")
    val fileName: String? = null,
    @SerializedName("uploader")
    val uploader: String? = null
    ): Serializable
